tests/generation/test_framework_agnostic.py

```diff
@@ -0,0 +1,41 @@
+"""
+Framework agnostic tests for generate()-related methods.
+"""
+
+import numpy as np
+
+from transformers import AutoTokenizer
+
+
+class GenerationIntegrationTestsMixin:
+
+    # To be populated by the child classes
+    framework_dependent_parameters = {
+        "AutoModelForSeq2SeqLM": None,
```

Comment:
I kind feel this should be something like `model_class: None`, and in `model_class: explicit class or auto calss` in subclasses.

But as you mentioned, this file is meant to be expanded, there is no real need to figure out what would be the best name at this moment.